### ISSUE TITLE
More work on holder binding

### ISSUE THREAD

**Issue by Sakurann** [oauthstuff/draft-selective-disclosure-jwt#24]
A comment from @b---c

> I'm a bit uneasy about how much binding there actually is in the holder binding. The whole SD-JWT + signed SD-JWT-R just becomes a bearer token. Oh, I now see https://www.ietf.org/archive/id/draft-fett-oauth-selective-disclosure-jwt-00.html#section-5-3.5.2.1.2.2.1 about the SD-JWT-R being bound to the current transaction. Seems more work is needed in this area. Like the empty sec 6.1 https://www.ietf.org/archive/id/draft-fett-oauth-selective-disclosure-jwt-00.html#name-holder-binding-2 :)

**Comment by Sakurann** [comment]
Yes, the holder binding as of now only achieves binding to the public key - the holder is only proving control over the same private key both during issuance and presentation. 

To prove that the actual user, about whom the user claims are, is in front of a device both during issuance and presentation, more is needed.
- one option would be to use a public key that is able to sign only after biometric matching is performed. But there are some limitations because mobile OS native mechanisms allow to register multiple biometrics for the same device.
- another option would be to include a portrait image as a claim in SD-JWT, holder app would compare it against a live image of the user, but will not release an SD-JWT claim in SD-JWT-R.

re transaction-binding, in the case of JWT-VCs, VP is bound to the transaction because it includes nonce and audience from the presentation request. How best this can be included in the draft, since req-res would be out of scope of SD-JWT draft?

**Comment by Sakurann** [comment]
my question is should the holder binding being made entirely out of scope of this draft? If not, suggest we make signing of SD-JWT-R mandatory. 

On one hand, as @christianpaquin has pointed out, when there is holder binding, what is the use-case where an SD-JWT-R is not signed? On another hand, even if SD-JWT-R is not signed (ie no holder binding), as long as the verifier performs verification, selective disclosure would work.

**Comment by danielfett** [comment]
I think we should speak about the concept of holder binding in this spec including pointing to the tools we provide to achieve this:

- Places where keys can potentially be communicated (in the SD-JWT!)
- The signature of the SD-JWT-R

We should **not** define how keys are created or distributed, the syntax of how they are communicated, how they are verified, or what their meaning can be (bound to a device? biometrics? etc.).

**Comment by selfissued** [comment]
This is too low level a spec to include anything about key distribution, etc.  That's a job for profiles.

By analogy, neither the JWS nor JWT specs said anything about where to get keys (by design!).  Whereas the ID Token definition in Connect, which is a profile, did exactly that.  Similar separation of responsibilities applies here.
